spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/MiniBatch.scala

```diff
@@ -95,11 +182,407 @@ class TensorMiniBatch[T](
   override def getTarget(): Activity = {
     target
   }
+
+  override def setValue(samples: Seq[Sample[T]])(implicit ev: TensorNumeric[T]): this.type = {
+    require(samples.length > 0, "samples is empty")
+    require(batchSize == 0 || samples.length <= batchSize, "setValue: samples's size doesn't " +
+      s"match mini batch size, excepted ${size()} got ${samples.length}")
+
+    if (featurePaddingParam.isDefined || labelPaddingParam.isDefined) {
+      val longestFeature = MiniBatch.findLongestFeatures(samples)
```

Comment:
Yes, we should.   featurePaddingParam.isDefined means samples's size is different.